test/framework/test_composite.py

```diff
@@ -82,23 +85,28 @@ def test_flatten_results_nested(self):
         comp_exp = ParallelExperiment(
             [
                 BatchExperiment(
-                    2 * [ParallelExperiment([exp0, exp1], flatten_results=False)],
-                    flatten_results=False,
+                    2 * [ParallelExperiment([exp0, exp1], flatten_results=True)],
+                    flatten_results=True,
                 ),
                 BatchExperiment(
-                    3 * [ParallelExperiment([exp2, exp3], flatten_results=False)],
-                    flatten_results=False,
+                    3 * [ParallelExperiment([exp2, exp3], flatten_results=True)],
+                    flatten_results=True,
                 ),
             ],
             flatten_results=True,
         )
         expdata = comp_exp.run(FakeBackend(num_qubits=4))
         self.assertExperimentDone(expdata)
         # Check no child data was saved
-        self.assertEqual(len(expdata.child_data()), 0)
+        self.assertEqual(len(expdata.child_data()), 2)
```

Comment:
Probably we can delete child containers when they are flattened, before going out from _run_analysis? This is technically breaking API change, i.e. we cannot change existing test within one release cycle.